python/cudf/cudf/tests/test_json.py

```diff
@@ -936,14 +936,35 @@ def test_json_dtypes_nested_data():
         """,
         ),
     ],
+    # TODO failing test cases
 )
-def test_order_nested_json_reader(tag, data):
-    expected = cudf.read_json(StringIO(data), engine="pandas", lines=True)
-    target = cudf.read_json(
-        StringIO(data), engine="cudf_experimental", lines=True
-    )
+class TestNestedJsonReaderCommon:
+    @pytest.mark.parametrize("chunk_size", [10, 100, 1024, 1024 * 1024])
+    def test_chunked_nested_json_reader(self, tag, data, chunk_size):
+        expected = pd.read_json(StringIO(data), lines=True)
+
+        source_size = len(data)
+        chunks = []
+        for chunk_start in range(0, source_size, chunk_size):
+            chunks.append(
+                cudf.read_json(
+                    StringIO(data),
+                    engine="cudf_experimental",
+                    byte_range=[chunk_start, chunk_size],
+                    lines=True,
+                )
+            )
+        df = cudf.concat(chunks).reset_index(drop=True)
+        assert_eq(df, expected, check_dtype=False)
```

Comment:
Why are we skipping the dtype check here? This may be okay, but we should probably add a comment with the justification.

Reply:
updated with py arrow table check